### PR TITLE
runtime-v2: log call stack on error

### PR DESCRIPTION
**Pull request by brig**
example logs:
```
14:38:08 [ERROR] (concord.yml): Error @ line: 20, col: 7. boom
14:38:08 [ERROR] Call stack:
(concord.yml) @ line: 15, col: 7, thread: 0, flow: flowB
(concord.yml) @ line: 6, col: 7, thread: 0, flow: flowA
```

I think we can remove column number from error messages, it's a bit useless. wdyt?

### CODE REVIEW

File: runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/StackTraceCollector.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            }
+            if (item == marker) {
+                return;
+            }
+        }
+    }
+
+//    @Override
+//    public Result onCommandError(Runtime runtime, VM vm, State state, ThreadId threadId, Command cmd) {
+        // clean stacktrace till exception handler (marker in before command for errorWrapper command);
```
Comment (ibodrov) [inline runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/StackTraceCollector.java:77]: Not needed?

Comment (brig) [reply runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/StackTraceCollector.java:77]: oops, thanks

File: runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/InMemoryState.java [hunk 2]
```diff
@@ -216,6 +217,59 @@ public Exception clearThreadError(ThreadId threadId) {
         }
     }
 
+    @Override
+    public List<StackTraceItem> getStackTrace(ThreadId threadId) {
+        synchronized (this) {
```
Comment (ibodrov) [inline runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/InMemoryState.java:222]: Change to
```java
public synchronized List<StackTraceItem> getStackTrace(ThreadId threadId)
```
?

`synchronized (this)` that spans the method's whole body is essentially just a synchronized method, AFAIR?

File: runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/StackTraceCollector.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import com.walmartlabs.concord.runtime.v2.runner.vm.FlowCallCommand;
+import com.walmartlabs.concord.svm.Runtime;
+import com.walmartlabs.concord.svm.*;
+
+public class StackTraceCollector implements ExecutionListener {
+
+    @Override
+    public Result afterCommand(Runtime runtime, VM vm, State state, ThreadId threadId, Command cmd) {
+        if (cmd instanceof FlowCallCommand) {
+            FlowCallCommand fcc = (FlowCallCommand)cmd;
```
Comment (ibodrov) [inline runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/StackTraceCollector.java:34]: Formatting: `(FlowCallCommand) cmd`

Comment (brig) [reply runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/StackTraceCollector.java]: fixed

File: runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/InMemoryState.java (hunk 2 above)
Comment (brig) [reply runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/InMemoryState.java:222]: yeah. same. but as you can see in this class all methods uses 
```
synchronized (this)
```
decided not to change the style :)

Comment (ibodrov) [reply runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/InMemoryState.java:222]: Yeah, not worth it.

**Review by ibodrov — APPROVED** [review]

**Pull request merged**